CONTRIBUTING.md

```diff
@@ -0,0 +1,139 @@
+
+## Contribution guidelines
+
+### Contributor’s Agreement
+
+To contribute source code to this repository, please read our [contributor's agreement](http://www.mulesoft.org/legal/contributor-agreement.html), and then execute it by running this notebook and following the instructions: https://api-notebook.anypoint.mulesoft.com/notebooks/#380297ed0e474010ff43
+
+### Pull requests are always welcome
+
+We are always thrilled to receive pull requests, and do our best to process them as fast as possible. Not sure if that typo is worth a pull request? Do it! We will appreciate it.
+
+If your pull request is not accepted on the first try, don't be discouraged! If there's a problem with the implementation, hopefully you received feedback on what to improve.
+
+We're trying very hard to keep the api-designer lean and focused. We don't want it to do everything for everybody. This means that we might decide against incorporating a new feature. However, there might be a way to implement that feature on top of api-designer.
+
+### Create issues...
+
+Any significant improvement should be documented as [a GitHub
+issue](https://github.com/mulesoft/api-designer/issues) before anybody
+starts working on it.
+
+### ...but check for existing issues first!
+
+Please take a moment to check that an issue doesn't already exist documenting your bug report or improvement proposal. If it does, it never hurts to add a quick "+1" or "I have this problem too". This will help prioritize the most common problems and requests.
+
+### Conventions
+
+Write clean code. Universally formatted code promotes ease of writing, reading, and maintenance. Make sure to avoid unnecessary white space changes which complicate diffs and make reviewing pull requests much more time consuming.
+
+Pull requests descriptions should be as clear as possible and include a reference to all the issues that they address. In other words, please try to explain in full detail the rationale for the change, and if the change is opaque (or hard) try to explain the fix as much as possible.  
+
+Pull requests must not contain commits from other users or branches.
+
+Commit messages must start with a short summary (max. 50 chars) written in the imperative, followed by an optional, more detailed explanatory text which is separated from the summary by an empty line.
+
+Code review comments may be added to your pull request. Discuss, then make the suggested modifications and push additional commits to your feature branch. Be sure to post a comment after pushing. The new commits will show up in the pull request automatically, but the reviewers will not be notified unless you comment.
+
+Before the pull request is merged, make sure that you squash your commits into logical units of work using git rebase -i and git push -f. After every commit, the test suite should be passing. Include documentation changes in the same commit so that a revert would remove all traces of the feature or fix.
```

Comment:
I'd go through and add code markup to code areas (partically `git rebase -i` and `git push -f`).